packages/shared/getComponentNameFromType.js

```diff
@@ -85,6 +86,8 @@ export default function getComponentNameFromType(type: mixed): string | null {
   }
   if (typeof type === 'object') {
     switch (type.$$typeof) {
+      case REACT_SERVER_CONTEXT_TYPE:
+        return ((type: any): ReactContext<any>)._globalName + '.Provider';
```

Comment:
If there is a `.displayName` we should probably use that like for other contexts and then fallback to `_globalName` if it doesn't exist. 

This should also be wrapped in the feature flag condition and moved to the bottom of the switch. That way it will fallthrough if the condition fails and therefore the case can be removed and therefore the symbol.